src/components/Hosts/UpdateStatusModal.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from "react";
+import { Select, Input } from "antd";
+import styled from "@emotion/styled";
+import { useMutation } from "@apollo/react-hooks";
+import { Modal } from "components/Modal";
+import { Button } from "components/Button";
+import {
+  UpdateHostStatusMutation,
+  UpdateHostStatusMutationVariables,
+} from "gql/generated/types";
+import { UPDATE_HOST_STATUS } from "gql/mutations";
+import { useBannerDispatchContext } from "context/banners";
+import { UpdateHostStatus } from "types/host";
+import { Body } from "@leafygreen-ui/typography";
+
+const { Option } = Select;
+const { TextArea } = Input;
+
+interface Props {
+  visible: boolean;
+  dataCy: string;
+  hostIds: string[];
+  closeModal: () => void;
+}
+
+export const UpdateStatusModal: React.FC<Props> = ({
+  visible,
+  dataCy,
+  hostIds,
+  closeModal,
+}) => {
+  const dispatchBanner = useBannerDispatchContext();
+
+  const [status, setHostStatus] = useState<UpdateHostStatus | null>(null);
+
+  const [notes, setNotesValue] = useState<string>("");
+
+  const resetForm = () => {
+    setHostStatus(null);
+    setNotesValue("");
+  };
+
+  // UPDATE HOST STATUS MUTATION
+  const [updateHostStatus, { loading: loadingUpdateHostStatus }] = useMutation<
+    UpdateHostStatusMutation,
+    UpdateHostStatusMutationVariables
+  >(UPDATE_HOST_STATUS, {
+    onCompleted({ updateHostStatus: numberOfHostsUpdated }) {
+      closeModal();
+      dispatchBanner.successBanner(
+        `Status was changed to ${status} for ${numberOfHostsUpdated} host${
+          numberOfHostsUpdated === 1 ? "" : "s"
+        }`
+      );
+      resetForm();
+    },
+    onError(error) {
+      closeModal();
+      dispatchBanner.errorBanner(
+        `There was an error updating hosts status: ${error}`
+      );
+    },
+    refetchQueries: ["Hosts"],
+  });
+
+  const onClickUpdate = () => {
+    updateHostStatus({ variables: { hostIds, status, notes } });
+  };
+
+  const onClickCancel = () => {
+    closeModal();
+    resetForm();
+  };
```

Comment:
nit: we can also inline these in the jsx components since they seem small enough and there isnt to much going on.

Reply:
`onClickCancel` is used in two different places and i prefer them this way